pkg/server/skydb/pq/builder/factory.go

```diff
@@ -326,3 +332,34 @@ type joinedTable struct {
 func (a joinedTable) equal(b joinedTable) bool {
 	return a.secondaryTable == b.secondaryTable && a.primaryColumn == b.primaryColumn && a.secondaryColumn == b.secondaryColumn
 }
+
+func (f *predicateSqlizerFactory) NewSort(s skydb.Sort) (string, error) {
```

Comment:
This function is doing the same thing as `SortOrderBySQL`, so you either need this `NewSort` or `SortOrderBySQL`. It is so unpredictable now, you cannot tell which code path will be executed for each type of expression. Please combine them into one single function.

Reply:
Done at 31bf84d